x/mongo/driver/connstring/connstring.go

```diff
@@ -466,6 +474,9 @@ func (p *parser) validateAuth() error {
 			return fmt.Errorf("SCRAM-SHA-256 cannot have mechanism properties")
 		}
 	case "":
+		if p.UsernameSet && p.Username == "" {
```

Comment:
Is a password also required if the user name is set?

Reply:
This spec change was specifically to ban empty usernames. I don't see any test coverage for a URI like `mongodb://username@localhost:27017`, but I think it should be invalid because the default mechanism is one of the SCRAM ones, which require both username and password. What do you think we should do here?

Reply:
As long as `mongodb://username@localhost:27017` doesn't panic, I'm ok leaving it like this.  It's not worth another spec ticket.